structure/deque/deque.go

```diff
@@ -0,0 +1,84 @@
+// description: Double Ended Queue is a generalized version of Queue data structure that allows insert and delete at both ends.
+// References:
+//	Wikipedia : https://en.wikipedia.org/wiki/Double-ended_queue
+//	Github: https://www.geeksforgeeks.org/deque-set-1-introduction-applications/
+// author [Sayan](https://github.com/bose-sayan)
+
+// Package deque implements a Double Ended Queue data structure.
+package deque
+
+import (
+	"errors"
+)
+
+// ErrEmptyDeQueue is a custom error for handling cases when some dequeuing operation is performed on an empty deque.
+var ErrEmptyDeQueue = errors.New("DoublyEnded queue is empty, so can't perform this operation")
+
+type DoublyEndedQueue[T any] struct {
+	deque []T
+}
+
+// NewDeque returns a new DoublyEndedQueue.
+func NewDeque[T any]() *DoublyEndedQueue[T] {
```

Comment:
```suggestion
func New[T any]() *DoublyEndedQueue[T] {
```

`deque` is in the name of the package so no need to repeat it here.